src/flow-spec/IBlock.ts

```diff
@@ -107,21 +101,62 @@ export function findBlockExitWith(uuid: string, block: IBlock): IBlockExit {
   return exit
 }
 
-export function findFirstTruthyEvaluatingBlockExitOn(block: IBlockWithTestExits, context: IContext): IBlockExitTestRequired | undefined {
+/**
+ * @param block
+ * @param context
+ * @deprecated Use firstTrueBlockExitOrThrow or firstTrueBlockExitOrNull
+ */
+export function findFirstTruthyEvaluatingBlockExitOn(block: IBlock, context: IContext): IBlockExit | undefined {
   const {exits} = block
   if (exits.length === 0) {
     throw new ValidationException(`Unable to find exits on block ${block.uuid}`)
   }
 
   const evalContext = createEvalContextFrom(context)
-  return find<IBlockExitTestRequired>(
-    exits,
-    ({test, default: isDefault = false}) => !isDefault && evaluateToBool(String(test), evalContext)
-  )
+  return find<IBlockExit>(exits, ({test, default: isDefault = false}) => !isDefault && evaluateToBool(String(test), evalContext))
+}
+
+export function firstTrueBlockExitOrNull(block: IBlock, context: IContext): IBlockExit | undefined {
+  try {
+    return firstTrueBlockExitOrThrow(block, context)
+  } catch (e) {
```

Comment:
Should we check if `e` is `ValidationException` and rethrow if it's not?

Reply:
For this one, no. The idea is that functions named `*OrNull()` will never throw an error. We are expecting that there can be an error, but we expect null to be returned if that case is encountered. This can lead to much cleaner code, that allows us to decide on the severity of encountering an error at the call site. If we want to consider an error to stop what we are doing, we would use the `*OrThrow()` functions. If we want to just lightly check for an exception, (such as in an if statement's expresssion, where we may be potentially expecting an error case) we can use the `*OrNull()` calls.